### DISCUSSION TITLE
Roadmap towards WebGPU support

### DISCUSSION THREAD

**Discussion by keller-mark** [visgl/deck.gl#7873] [General]
Hi- I am curious whether DeckGL has a public roadmap towards incorporating WebGPU as it is starting to move from its experimental status.

I see it in a few [issues](https://github.com/search?q=repo%3Avisgl%2Fdeck.gl+webgpu&type=issues), [pull requests](https://github.com/search?q=repo%3Avisgl%2Fdeck.gl+webgpu&type=pullrequests), and [discussions](https://github.com/search?q=repo%3Avisgl%2Fdeck.gl+webgpu&type=discussions), but it seems to just be mentioned in passing without wider context. It would be great to have a better sense of whether this is a priority, and if so, perhaps the incremental deliverables could be described in an [RFC](https://github.com/visgl/deck.gl/tree/master/dev-docs/RFCs/proposals) to understand how far off it might be.

I know it may be more of a question for libraries such as https://github.com/visgl/luma.gl and in that case even just pointers to their respective WebGPU roadmaps (if they exist) would be great!

**Comment by ibgreen** [comment]
@keller-mark Thanks for your interest. 

The technical approach is defined and a substantial amount of work has already been done on luma.gl v9 which now offers a WebGPU style API and supports both WebGL and WebGPU. 

There is a deck.gl v9 development branch `9.0-dev` that integrates luma.gl v9. There is ongoing work around promoting that to the master branch.

My sense is that things will play out like this:
- For deck.gl v9 we will likely upgrade to luma.gl v9 API without actually enabling WebGPU, and focus on getting deck.gl to work on the new WebGPU style luma API on the WebGL backend, and minimizing any breaking changes in the deck.gl API.
- And then enable we will WebGPU in deck.gl v10 and focus on interoperability and maximizing performance gains from WebGPU.

Still speculating (assuming current maintainers only, no additional contributors):
- v9 will release end of Q3
- v10 with WebGPU perhaps in Q1 next year.

Naturally, additional contributions could substantially accelerate this.

**Reply by keller-mark** [reply]
Thanks for the quick reply, this is great to know! Thanks for all of your work on deck.gl!
